apps/chat/src/components/AppsEditor/Settings/index.tsx

```diff
@@ -105,11 +104,20 @@ export const ApplicationSettings: React.FC<Props> = ({
   const isCodeEditorDirty = useAppSelector(CodeEditorSelectors.selectIsDirty);
   const theme = useAppSelector(UISelectors.selectThemeState);
 
-  const [previewMode, setPreviewMode] = useState<PreviewMode>(
-    schema?.[ApplicationTypeSchemaProperties.applicationTypeViewerUrl]
+  const [previewMode, setPreviewMode] = useState<PreviewMode>(() => {
+    if (screenState === ScreenState.MD) return PreviewMode.closed;
+    return schema?.[ApplicationTypeSchemaProperties.applicationTypeViewerUrl]
       ? PreviewMode.closed
-      : PreviewMode.half,
-  );
+      : PreviewMode.half;
+  });
+
+  useEffect(() => {
+    if (screenState === ScreenState.MD) {
```

Comment:
we have `full` and `closed` for all devices on Settings step, only `half` for tablet(`MD`) and for mobile(`SM`) should be reset to `closed`
```suggestion
    if (screenState <= ScreenState.MD && previewMode === PreviewMode.half) {
        setPreviewMode(PreviewMode.closed);
    }
```
keep in mind that `screenState` can be `SM`(768) and I can switch from XL to SM directly